subsys/usb/class/bluetooth.c

```diff
@@ -66,6 +66,10 @@ NET_BUF_POOL_DEFINE(acl_rx_pool, ACL_BUF_COUNT, BT_BUF_ACL_SIZE,
 #define BLUETOOTH_OUT_EP_ADDR		0x02
 #define BLUETOOTH_IN_EP_ADDR		0x82
 
+/* TODO: Replace use of USB_MAX_FS_INT_MPS if higher speeds are supported */
```

Comment:
This seems to be a limitation on USB stack, so we can't set values bigger than supported by the current speed as the stack would not truncate and expose the value as is to the host causing warning such as:

[50969.954964] usb 1-3: config 1 interface 0 altsetting 0 endpoint 0x1 has invalid maxpacket 107, setting to 64
[50969.954967] usb 1-3: config 1 interface 0 altsetting 0 endpoint 0x82 has invalid maxpacket 107, setting to 64

Reply:
Afaik fragmentation is already handled by HCI layer so we don't have to do in USB again.

Reply:
Let me take this back, it does seem that USB itself would need to handle fragmentation somehow because the HCI would fragment accourding to the HCI buffer size not the USB MPS, @jhedberg @holtmann or do we actually break the fragments accourding to the MPS on the drivers? In any case there doesn't seem to be any header that would make reassemble possible, so we can't really tell if that is a fragment of a full packet, beside on TX path we don't actually attempt to fragment either so I assume the USB core would have to deal with that. We could also force disable the fragmentation at USB by forcing HCI buffer to be <= 64.

Reply:
There is BLUETOOTH CORE SPECIFICATION Version 5.2 | Vol 4, Part B which talks about USB frames and sizes. For now I think we can postpone this for this PR. Limiting the HCI buffer to be <= 64. Which would mean limiting the controller TX/RX buffer, CONFIG_BT_CTLR_TX_BUFFERS/CONFIG_BT_CTLR_RX_BUFFERS. It would limiting us for data length extension and advertising extension for the time being.